CHANGELOG.md

```diff
@@ -1,5 +1,72 @@
 # Changelog
 
+## 0.0.1-alpha.9
+
+### Typing semantics and features
+
+- Add generic inference for dataclasses ([#18443](https://github.com/astral-sh/ruff/pull/18443))
+- Add support for global `__debug__` constant ([#18540](https://github.com/astral-sh/ruff/pull/18540))
+- Argument type expansion for overload call evaluation ([#18382](https://github.com/astral-sh/ruff/pull/18382))
+- Exclude members starting with `_abc_` from a protocol interface ([#18467](https://github.com/astral-sh/ruff/pull/18467))
+- Infer `list[T]` for starred target in unpacking ([#18401](https://github.com/astral-sh/ruff/pull/18401))
+- Infer `list[T]` when unpacking non-tuple type ([#18438](https://github.com/astral-sh/ruff/pull/18438))
+- Support type annotation for legacy typing aliases for generic classes ([#18404](https://github.com/astral-sh/ruff/pull/18404))
+- Allow using `dataclasses.dataclass` as a function ([#18440](https://github.com/astral-sh/ruff/pull/18440))
+- Type narrowing for attribute/subscript assignments ([#18041](https://github.com/astral-sh/ruff/pull/18041))
+
+### Diagnostics
+
+- Add hints to `invalid-type-form` for common mistakes ([#18543](https://github.com/astral-sh/ruff/pull/18543))
+- Add subdiagnostic suggestion to `unresolved-reference` diagnostic when variable exists on `self` ([#18444](https://github.com/astral-sh/ruff/pull/18444))
+- Track the origin of the `environment.python` setting for better error messages ([#18483](https://github.com/astral-sh/ruff/pull/18483))
+
+### CLI
+
+- Fix `--python` argument for Windows, and improve error messages for bad `--python` arguments ([#18457](https://github.com/astral-sh/ruff/pull/18457))
+
+### Bug fixes
+
+- Meta-type of type variables should be `type[..]` ([#18439](https://github.com/astral-sh/ruff/pull/18439))
+- Only consider a type `T` a subtype of a protocol `P` if all of `P`'s members are fully bound on `T` ([#18466](https://github.com/astral-sh/ruff/pull/18466))
+- Fix false positives for legacy `ParamSpec`s inside `Callable` type expressions ([#18426](https://github.com/astral-sh/ruff/pull/18426))
+- Fix panic when pulling types for `UnaryOp` expressions inside `Literal` slices ([#18536](https://github.com/astral-sh/ruff/pull/18536))
+- Fix panic when trying to pull types for attribute expressions inside `Literal` type expressions ([#18535](https://github.com/astral-sh/ruff/pull/18535))
+- Fix panic when trying to pull types for subscript expressions inside `Callable` type expressions ([#18534](https://github.com/astral-sh/ruff/pull/18534))
```

Comment:
these three could possibly go under the "server" section, since I believe users would have been most likely to encounter them if they hovered over one of these AST nodes in an IDE context

Reply:
I'm not sure if it makes sense to put them under "server" but I'm also not sure if they belong in the changelog because these were panics in our testing files. I'm fine with removing them but thought to put them under "bugs" in case anyone was facing them.

Reply:
The panic was _uncovered_ by improving our test coverage, but it could easily have been triggered by a user hovering over exactly the wrong AST node in their IDE. Here's an example of exactly such a crash that was previously reported by a user: https://github.com/astral-sh/ty/issues/366#issuecomment-2909869832. It was fixed in https://github.com/astral-sh/ruff/pull/18321. So I think these definitely deserve to be listed in the changelog!

Reply:
Oh, I see. Thanks for the context!